### PR TITLE
pythonPackages.python-osc: init at 1.7.7

### PR DESCRIPTION
**Pull request by anirrudh**
###### Motivation for this change

Porting python-osc for some projects: https://github.com/attwad/python-osc

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 139252` at 36f42274 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.python-osc</li>
    <li>python39Packages.python-osc</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `python`.
Near pkgs/development/python-modules/python-osc/default.nix:4:3:
```
  |
4 | , python
  |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/python-osc/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    sha256 = "4c7d68a7719d9425ab2a4ee9a2b9d5a9f5b66593fb46e20e38f91e1452bea2d2";
+  };
+
+  pythonImportsCheck = [ "pythonosc" ];
+
+  meta = with lib; {
+    description = "Open Sound Control server and client in pure python";
+    homepage = "https://github.com/attwad/python-osc";
+    license = licenses.unlicense;
+    maintainers = with maintainers; [ "anirrudh" ];
```
Comment (figsoda) [inline pkgs/development/python-modules/python-osc/default.nix:22]:
```suggestion
    maintainers = with maintainers; [ anirrudh ];
```

File: pkgs/development/python-modules/python-osc/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, python
```
Comment (figsoda) [inline pkgs/development/python-modules/python-osc/default.nix:4]:
```suggestion
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -736,6 +736,12 @@
     githubId = 1771266;
     name = "Vo Anh Duy";
   };
+  anirrudh = {
```
Comment (figsoda) [inline maintainers/maintainer-list.nix:739]: can you it a separate commit: `maintainers: add anirrudh` that goes before your current commit?

Comment (anirrudh) [reply maintainers/maintainer-list.nix:739]: Yep! I apologize - I squashed everything into one!

**Review by fabaff — COMMENTED** [review]
The upstream repo contains [tests](https://github.com/attwad/python-osc/tree/master/pythonosc/test). They are part of the releases on GitHub (-> use `fetchFromGitHub` instead of `fetchPypi`).

Could you check if those can be executed?

**Comment by anirrudh** [comment]
Absolutely! I'll build a new derivation and PR it in with tests running tomorrow.

**Comment by costrouc** [comment]
Result of `nixpkgs-review pr 139252` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.python-osc</li>
    <li>python39Packages.python-osc</li>
  </ul>
</details>

**Comment by figsoda** [comment]
@fabaff I checked the output of `fetchPypi`, can confirm tests are included

**Comment by figsoda** [comment]
@ofborg build python3Packages.python-osc

File: pkgs/development/python-modules/python-osc/default.nix (hunk 1 above)
Comment (figsoda) [reply pkgs/development/python-modules/python-osc/default.nix:21]: I think you forgot to change it

Comment (anirrudh) [reply pkgs/development/python-modules/python-osc/default.nix:21]: Ah, I did, my bad. Shall commit now!

Comment (anirrudh) [reply pkgs/development/python-modules/python-osc/default.nix]: Resolved!

**Comment by anirrudh** [comment]
@fabaff with the recent update from @figsoda, do you still think fetchFromGithub is required?

**Review by figsoda — APPROVED** [review]
lgtm, I don't think `fetchFromGitHub` is needed any more

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 139252` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.python-osc</li>
    <li>python39Packages.python-osc</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**